### PR TITLE
uninstall flow for MCG

### PR DESCRIPTION
**Pull request by naveenpaul1**
addon uninstall flow

//cc : @rexagod @v-harihar @aspandey

### CODE REVIEW

File: controllers/managedmcg_controller.go [hunk 1]
```diff
@@ -177,25 +183,55 @@ func (r *ManagedMCGReconciler) Reconcile(ctx context.Context, req ctrl.Request)
...
+
+			// Deleting Noobaa CSV from the namespace
+			/*r.Log.Info("deleting Noobaa CSV")
+			if csv, err := r.getCSVByPrefix(noobaaOperatorName); err == nil {
+				if err := r.delete(csv); err != nil {
+					return ctrl.Result{}, fmt.Errorf("unable to delete csv: %v", err)
+				}
+			} else {
+				return ctrl.Result{}, err
+			}*/
```
Comment (rexagod) [inline controllers/managedmcg_controller.go:203]: We do need to delete the CSV as well, right?

File: controllers/managedmcg_controller.go [hunk 2]
```diff
@@ -215,30 +251,87 @@ func (r *ManagedMCGReconciler) reconcilePhases() (reconcile.Result, error) {
...
+	}
+}
+
+func (r *ManagedMCGReconciler) areComponentsReadyForUninstall() bool {
+	subComponents := r.managedMCG.Status.Components
+	return subComponents.Noobaa.State == mcgv1alpha1.ComponentReady
+}
+
+func (r *ManagedMCGReconciler) checkUninstallCondition() bool {
+	configmap := &corev1.ConfigMap{}
```
Comment (rexagod) [inline controllers/managedmcg_controller.go:320]: What are we using this ConfigMap for? Is there anything being stored in it? Do we really need this?

**Review by rexagod — CHANGES_REQUESTED** [review]

File: controllers/managedmcg_controller.go (hunk 2 above)
Comment (naveenpaul1) [reply controllers/managedmcg_controller.go:304]: @rexagod  This ConfigMap is created from addon when we click on the uninstall button in OCM, and its like signal us to delete the resources, It doesn't have any data but annotations

File: controllers/managedmcg_controller.go (hunk 1 above)
Comment (naveenpaul1) [reply controllers/managedmcg_controller.go]: @rexagod  we need to delete mcg CSV and we are doing it in removeOLMComponents() method

Comment (rexagod) [reply controllers/managedmcg_controller.go]: Ah okay, in that case can we remove this comment?

File: controllers/managedmcg_controller.go (hunk 2 above)
Comment (rexagod) [reply controllers/managedmcg_controller.go:304]: Hmm, okay.

**Review by rexagod — APPROVED** [review]

**Review by v-harihar — APPROVED** [review]
/lgtm

**Pull request merged**